Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Controllers/MixedRealityControllerVisualizer.cs

```diff
@@ -0,0 +1,285 @@
+            base.OnEnable();
+
+            if (MixedRealityManager.Instance.ActiveProfile.ControllersProfile != null && MixedRealityManager.Instance.ActiveProfile.ControllersProfile.RenderMotionControllers)
+            {
+                CreateControllerInstance(Handedness.Left);
+                CreateControllerInstance(Handedness.Right);
+            }
+        }
+
+        /// <summary>
+        /// If the visualizer is disabled, ensure all rendered controllers are cleaned up
+        /// </summary>
+        protected override void OnDisable()
+        {
+            DestroyControllerInstance(Handedness.Left);
+            DestroyControllerInstance(Handedness.Right);
+
+            base.OnDisable();
+        }
+
+        #endregion Enable / Disable
+
+        #region Controller Visualization Functions
+
+        /// <summary>
+        /// Get the currently configured model for the controller
+        /// </summary>
+        /// <remarks>
+        /// Controller is detected in the following order:
+        /// 1: The controller model attached to the specific controller type's configuration profile
+        /// 2: The generic controller model attached to the main controller configuration profile
+        /// </remarks>
+        /// <param name="sourceController"></param>
+        /// <param name="controllerModel"></param>
+        /// <returns>Returns true if a controller model is found and outputs the GameObject definition.  if no controller is found, the response is false</returns>
+        public bool TryGetControllerModel(IMixedRealityController sourceController, out GameObject controllerModel, out MixedRealityPose poseOffset)
+        {
+            controllerModel = null;
+            poseOffset = MixedRealityPose.ZeroIdentity;
+
+            //Try and get the controller model from the specific COntroller definition
+            MixedRealityManager.Instance.ActiveProfile.ControllersProfile?.MixedRealityControllerMappingProfiles.GetControllerModelOverride(sourceController.GetType(), sourceController.ControllerHandedness, out controllerModel, out poseOffset);
+            if (controllerModel != null)
+            {
+                return true;
+            }
+
+            //If no specific controller model found for the device type, try and get the generic override models from the main Controllers profile
+            if (sourceController.ControllerHandedness == Handedness.Left && MixedRealityManager.Instance.ActiveProfile.ControllersProfile?.GlobalLeftHandModel != null)
+            {
+                controllerModel = MixedRealityManager.Instance.ActiveProfile.ControllersProfile.GlobalLeftHandModel;
+                poseOffset = MixedRealityManager.Instance.ActiveProfile.ControllersProfile.LeftHandModelPoseOffset;
+                return true;
+            }
+
+            if (sourceController.ControllerHandedness == Handedness.Right && MixedRealityManager.Instance.ActiveProfile.ControllersProfile?.GlobalRightHandModel != null)
```

Comment:
else if?

Reply:
No, it's a fall though case, no need for an else.  Plus with the "Return" the function terminates, so an else would be meaningless

Reply:
more proof that I need new glasses... :\